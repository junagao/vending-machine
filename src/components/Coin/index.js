export { default as CoinItem } from './CoinItem';
export { default as CoinList } from './CoinList';
export { default as CoinSlot } from './CoinSlot';
export { default as CollectRefund } from './CollectRefund';
export { default as DisplayDepositedAmount } from './DisplayDepositedAmount';
export { default as RefillCoins } from './RefillCoins';
export { default as Wallet } from './Wallet';
