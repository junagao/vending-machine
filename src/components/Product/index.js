export { default as CollectProduct } from './CollectProduct';
export { default as ProductList } from './ProductList';
export { default as ProductItem } from './ProductItem';
export { default as RefillProductStock } from './RefillProductStock';
