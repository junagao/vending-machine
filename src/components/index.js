export { default as CoinSlot } from 'components/CoinSlot';
export { default as CollectProduct } from 'components/CollectProduct';
export { default as VendingMachine } from 'components/VendingMachine';
export { default as VendingMachineScreen } from 'components/VendingMachineScreen';
export { default as Wallet } from 'components/Wallet';
export { default as ProductList } from 'components/ProductList';
export { default as Header } from 'components/Header';
export { default as CollectRefund } from 'components/CollectRefund';
