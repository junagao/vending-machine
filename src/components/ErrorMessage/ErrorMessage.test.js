import React from 'react';
import { shallow } from 'enzyme';
import { ErrorMessage } from 'components';

describe('ErrorMessage', () => {
  let wrapper;
  const errorMessageMock = 'ERROR';

  beforeEach(() => {
    wrapper = shallow(<ErrorMessage errorMessage={errorMessageMock} />);
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one ErrorContainer button element', () => {
    expect(wrapper.find('ErrorContainer').length).toEqual(1);
  });

  it('should render the error message', () => {
    expect(
      wrapper
        .find('ErrorContainer')
        .children()
        .text(),
    ).toEqual('ERROR');
  });
});
