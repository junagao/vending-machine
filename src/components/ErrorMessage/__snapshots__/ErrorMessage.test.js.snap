// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorMessage should render correctly given the required props 1`] = `
<ErrorContainer>
  ERROR
</ErrorContainer>
`;
